src/Features/Core/Portable/Completion/CompletionServiceWithProviders.cs

```diff
@@ -69,117 +48,31 @@ public override CompletionRules GetRules()
         protected virtual ImmutableArray<CompletionProvider> GetBuiltInProviders()
             => ImmutableArray<CompletionProvider>.Empty;
 
-        private IEnumerable<Lazy<CompletionProvider, CompletionProviderMetadata>> GetImportedProviders()
-        {
-            if (_importedProviders == null)
-            {
-                var language = Language;
-                var mefExporter = (IMefHostExportProvider)_workspace.Services.HostServices;
-
-                var providers = ExtensionOrderer.Order(
-                        mefExporter.GetExports<CompletionProvider, CompletionProviderMetadata>()
-                        .Where(lz => lz.Metadata.Language == language)
-                        ).ToList();
-
-                Interlocked.CompareExchange(ref _importedProviders, providers, null);
-            }
-
-            return _importedProviders;
-        }
-
-        private ImmutableArray<CompletionProvider> CreateRoleProviders(ImmutableHashSet<string> roles)
-        {
-            var providers = GetAllProviders(roles);
-
-            foreach (var provider in providers)
-            {
-                _nameToProvider[provider.Name] = provider;
-            }
-
-            return providers;
-        }
-
-        private ImmutableArray<CompletionProvider> GetAllProviders(ImmutableHashSet<string> roles)
-        {
-            var imported = GetImportedProviders()
-                .Where(lz => lz.Metadata.Roles == null || lz.Metadata.Roles.Length == 0 || roles.Overlaps(lz.Metadata.Roles))
-                .Select(lz => lz.Value);
+        protected ImmutableArray<CompletionProvider> GetProviders(ImmutableHashSet<string> roles)
+            => _providerSource.GetProvidersAsync(roles ?? ImmutableHashSet<string>.Empty).Result;
 
-#pragma warning disable 0618
-            // We need to keep supporting built-in providers for a while longer since this is a public API.
-            // https://github.com/dotnet/roslyn/issues/42367
-            var builtin = GetBuiltInProviders();
-#pragma warning restore 0618
+        [Obsolete("Please use GetProvidersAsync instead.")]
```

Comment:
I think it's safe to break this since I can't find anyone use this method 
http://index/?leftProject=Microsoft.CodeAnalysis.Features&leftSymbol=a3tjdtcvghks
https://grep.app/search?q=CompletionServiceWithProviders